test/distributed/test_multi_threaded_pg.py

```diff
@@ -38,6 +39,21 @@ def test_broadcast_object_list(self):
         dist.broadcast_object_list(object_list=object_list)
         self.assertEqual(99, object_list[0])
 
+    def test_all_gather_object(self):
+        output = [None] * dist.get_world_size()
+        dist.all_gather_object(
+            object_list=output,
+            obj=dist.get_rank())
+
+        for i, v in enumerate(output):
+            self.assertEqual(i, v, f"rank: {dist.get_rank()}")
+
+    def test_all_reduce(self):
+        output = torch.ones(3, 3) * dist.get_rank()
+        dist.all_reduce(output)
+        res_num = ((0 + self.world_size - 1) * self.world_size) / 2
+        self.assertEqual(output, torch.ones(3, 3) * res_num)
+
     def test_something_else(self):
```

Comment:
Noob question: do you know what is this for?

Reply:
I deleted this, looks useless lol

Reply:
@wz337 It's the guidance left by @kumpera for actual n00bs who don't know where to add more tests like me. ;-)

Reply:
@XilunWu Haha thanks for providing the context. Otherwise, I would spend a long time wondering what kind of dark magic Rodrigo is trying to do here. Thanks for debunking the myth. =)